src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -36,6 +38,155 @@ Pane::Pane(const GUID& profile, const TermControl& control, const bool lastFocus
+    }
+    else if (_splitState == SplitState::Horizontal)
+    {
+        const auto paneSizes = _GetPaneSizes(height);
+
+        Size firstSize{ width, paneSizes.first };
+        Size secondSize{ width, paneSizes.second };
+        _firstChild->ResizeContent(firstSize);
+        _secondChild->ResizeContent(secondSize);
+    }
+}
+
+// Method Description:
+// - Adjust our child percentages to increase the size of one of our children
+//   and decrease the size of the other.
+// - Adjusts the separation amount by 5%
+// Arguments:
+// - direction: the direction to move our separator. If it's down or right,
+//   we'll be increasing the size of the first of our children. Else, we'll be
+//   decreasing the size of our first child.
+// Return Value:
+// - <none>
+void Pane::_DoResize(const Direction& direction)
+{
+    if (!DirectionMatchesSplit(direction, _splitState))
+    {
+        return;
+    }
+
+    float amount = .05f;
+    if (direction == Direction::Right || direction == Direction::Down)
+    {
+        amount *= -1.0f;
+    }
+
+    // Make sure we're not making a pane explode here by resizing it to 0
+    _firstPercent = std::clamp(_firstPercent.value() - amount, .05f, .95f);
+    // Update the other child to fill the remaining percent
+    _secondPercent = 1.0f - _firstPercent.value();
+
+    // Resize our columns to match the new percentages.
+    const Size actualSize{ gsl::narrow_cast<float>(_root.ActualWidth()),
+                           gsl::narrow_cast<float>(_root.ActualHeight()) };
+    ResizeContent(actualSize);
+}
+
+// Method Description:
+// - Moves the separator between panes, as to resize each child on either size
+//   of the separator. Tries to move a separator in the given direction. The
+//   separatior moved is the separator that's closest depth-wise to the
+//   currently focused pane, that's also in the correct direction to be moved.
+//   If there isn't such a separator, then this method returns false, as we
+//   couldn't handle the resize.
+// Arguments:
+// - direction: The direction to move the separator in.
+// Return Value:
+// - true if we or a child handled this resize request.
+bool Pane::ResizePane(const Direction& direction)
+{
+    // If we're a leaf, do nothing. We can't possibly have a descentant with a
+    // separator the correct direction.
+    if (_IsLeaf())
+    {
+        return false;
+    }
+
+    // Check if either our first or second child is the currently focused leaf.
+    // If it is, and the requested resize direction matches our separator, then
+    // we're the pane that needs to adjust its separator.
+    // If our separator is the wrong direction, then we can't handle it.
+    const bool firstIsFocused = _firstChild->_IsLeaf() && _firstChild->_lastFocused;
+    const bool secondIsFocused = _secondChild->_IsLeaf() && _secondChild->_lastFocused;
+    if (firstIsFocused || secondIsFocused)
+    {
+        if (Pane::DirectionMatchesSplit(direction, _splitState))

```

Comment:
>Pane:: [](start = 12, length = 6)

nit: do we need the `Pane::` prefix here?

Reply:
Apparently not